### PR TITLE
Move `register_views` to auth manager interface

### PR DESCRIPTION
**Pull request by vincbeck**
Auth managers have the possibility of implementing their own views. To do that, it had to be done in the security manager override. The security manager override had been creating to override/customize the security manager. This is something needed by the FAB auth manager, but other than that, the other auth managers should not have the need of creating a security manager override.

To avoid creating security manager override just for the sake of registering views specific to auth manager, moving this method to the interface will simplify things.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by jscheffl** [comment]
I see the interfaces with this PR are much simpler. But for the coupling of Providers and Core, this smells like a breaking change.
Do you have a proposal how to further release updates for provider compatible with Airflow 2.8/2.9/2.10?
Is the core change intended to be for Airflow 3?

**Comment by vincbeck** [comment]
> I see the interfaces with this PR are much simpler. But for the coupling of Providers and Core, this smells like a breaking change. Do you have a proposal how to further release updates for provider compatible with Airflow 2.8/2.9/2.10? Is the core change intended to be for Airflow 3?

Yes you are right, I should have mentioned it in the PR. I'll take care of that today

**Comment by vincbeck** [comment]
> Is the core change intended to be for Airflow 3?

Yes

**Comment by vincbeck** [comment]
> Do you have a proposal how to further release updates for provider compatible with Airflow 2.8/2.9/2.10?

I dont think this is a problem. The newly created method `register_views` in FAB auth manager calls underneath `register_views` in the security manager. Airflow 2 calls `register_views` from the security manager. Therefore, any modifications in the `register_views` method from security manager will be available in Airflow 2 and 3

The issue is all the way around, making sure that Airflow 3 can use older version of FAB provider. I'll handle it

**Comment by vincbeck** [comment]
I resolved the backward incompatibility issue. It should be all fine now.

File: airflow/auth/managers/base_auth_manager.py [hunk 1]
```diff
@@ -443,3 +430,19 @@ def security_manager(self) -> AirflowSecurityManagerV2:
...
+
+        Override this method to expose commands via Airflow CLI to manage this auth manager.
+        """
+        return []
+
+    def get_api_endpoints(self) -> None | Blueprint:
+        """Return API endpoint(s) definition for the auth manager."""
+        return None
+
+    def register_views(self) -> None:
```
Comment (jscheffl) [inline airflow/auth/managers/base_auth_manager.py:447]: This method does not carry any parameter and has no return. The only way to "register" something is - as seen in the AWS auth mgr - to use the FAB appbuilder class from the global app variable.

If we do it like this, we can not deprecate and make FAB optional. Is there any option not adding a new binding to FAB when we add the UI registration? Bare Flask endpoints/routes and menu names?

Comment (jscheffl) [reply airflow/auth/managers/base_auth_manager.py:447]: FYI @bbovenzi or @pierrejeambrun this smells a bit like the idea of AIP-68 but for another purpose - do you have an idea how to expose "views" w/o destroying the future UI concept in React? (w/o having the new plugin mechanism in place today and and not adding a new binding to FAB?

**Review by jscheffl — COMMENTED** [review]
Did a review - if we would not want to replace FAB in Airflow 3 I would have made an approve, but as we actually want to get rid of it... Might be we need to introduce another method of integration that is future proof.

Will the AWS Auth Manager render the views inside Airflow UI or just add menu entries which point to an external site?

**Comment by vincbeck** [comment]
> Did a review - if we would not want to replace FAB in Airflow 3 I would have made an approve, but as we actually want to get rid of it... Might be we need to introduce another method of integration that is future proof.
> 
> Will the AWS Auth Manager render the views inside Airflow UI or just add menu entries which point to an external site?

100% agree with you but the intent of this PR is not to drop FAB from Airflow. That requires a lot of work and will be done in separate PRs. The way I register views in the auth manager is the way it is done today so I dont add anything new. Definitely this will need to be reworked/redone later to complete AIP-79. And yes likely we will need a mechanism similar to AIP-68 will expose for plugins so that the auth manager can add new views to an Airflow environment

File: airflow/auth/managers/base_auth_manager.py (hunk 1 above)
Comment (vincbeck) [reply airflow/auth/managers/base_auth_manager.py:447]: As mentioned [here](https://github.com/apache/airflow/pull/41777#issuecomment-2316299463), this is not in the scope of this PR

**Comment by jscheffl** [comment]
> > Did a review - if we would not want to replace FAB in Airflow 3 I would have made an approve, but as we actually want to get rid of it... Might be we need to introduce another method of integration that is future proof.
> > Will the AWS Auth Manager render the views inside Airflow UI or just add menu entries which point to an external site?
> 
> 100% agree with you but the intent of this PR is not to drop FAB from Airflow. That requires a lot of work and will be done in separate PRs. The way I register views in the auth manager is the way it is done today so I dont add anything new. Definitely this will need to be reworked/redone later to complete AIP-79. And yes likely we will need a mechanism similar to AIP-68 will expose for plugins so that the auth manager can add new views to an Airflow environment

I just want to clarify not to add a new interface and therefore add a new legacy - even if it is used the same way but unclean today.

**Comment by vincbeck** [comment]
> > > Did a review - if we would not want to replace FAB in Airflow 3 I would have made an approve, but as we actually want to get rid of it... Might be we need to introduce another method of integration that is future proof.
> > > Will the AWS Auth Manager render the views inside Airflow UI or just add menu entries which point to an external site?
> > 
> > 
> > 100% agree with you but the intent of this PR is not to drop FAB from Airflow. That requires a lot of work and will be done in separate PRs. The way I register views in the auth manager is the way it is done today so I dont add anything new. Definitely this will need to be reworked/redone later to complete AIP-79. And yes likely we will need a mechanism similar to AIP-68 will expose for plugins so that the auth manager can add new views to an Airflow environment
> 
> I just want to clarify not to add a new interface and therefore add a new legacy - even if it is used the same way but unclean today.

Nop, this already exist today. I just moved some code from security manager to auth manager

**Comment by jscheffl** [comment]
> > Did a review - if we would not want to replace FAB in Airflow 3 I would have made an approve, but as we actually want to get rid of it... Might be we need to introduce another method of integration that is future proof.
> > Will the AWS Auth Manager render the views inside Airflow UI or just add menu entries which point to an external site?
> 
> 100% agree with you but the intent of this PR is not to drop FAB from Airflow. That requires a lot of work and will be done in separate PRs. The way I register views in the auth manager is the way it is done today so I dont add anything new. Definitely this will need to be reworked/redone later to complete AIP-79. And yes likely we will need a mechanism similar to AIP-68 will expose for plugins so that the auth manager can add new views to an Airflow environment

I just want to clarify not to add a new interface and therefore add a new legacy - even if it is used the same way but unclean today.

**Review by o-nikolas — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Pull request merged**